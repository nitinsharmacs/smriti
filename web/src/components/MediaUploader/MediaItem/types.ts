import type {
  MediaType,
  MediaUploadStatus,
} from 'src/components/MediaUploader/types';

export interface MediaItemProps {
  name: string;
  type: MediaType;
  status: MediaUploadStatus;
  progress?: number; // valid for media status "InProgress"
}
