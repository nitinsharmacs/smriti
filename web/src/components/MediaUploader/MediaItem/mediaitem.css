.upload-media-item {
  display: flex;
  align-items: center;
  background-color: var(--main-bg-color);
  padding: 1em;
  border-radius: 4px;
  margin: 0.5em 0;
}

.media-icon,
.media-name,
.media-status {
  line-height: 0;
}

.media-name {
  margin-left: 1em;
}

.media-status {
  margin-left: auto;
}

.media-status > button {
  padding: 0;
  color: var(--main-text-fg-color);
}
