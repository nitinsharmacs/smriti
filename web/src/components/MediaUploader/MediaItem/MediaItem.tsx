import InsertPhotoOutlinedIcon from '@mui/icons-material/InsertPhotoOutlined';
import VideocamOutlinedIcon from '@mui/icons-material/VideocamOutlined';
import CircularProgressBar from 'src/components/CircularProgress/CircularProgress';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import ReplayIcon from '@mui/icons-material/Replay';
import ErrorIcon from '@mui/icons-material/Error';

import { type MediaItemProps } from 'src/components/MediaUploader/MediaItem/types';
import { MediaType, MediaUploadStatus } from '../types';
import './mediaitem.css';
import { IconButton } from '@mui/material';
import { doNothing } from 'src/helpers';

const MediaStatus = (
  status: MediaUploadStatus,
  progress: number = 0,
  onRetry: () => void = doNothing
) => {
  switch (status) {
    case MediaUploadStatus.InProgress:
      return <CircularProgressBar value={progress} size={20} thickness={8} />;
    case MediaUploadStatus.Success:
      return <CheckCircleIcon />;
    case MediaUploadStatus.Failed:
      return (
        <IconButton onClick={onRetry}>
          <ReplayIcon />
        </IconButton>
      );
    default:
      return <ErrorIcon />;
  }
};

const MediaItem = (props: MediaItemProps) => {
  return (
    <div className='upload-media-item'>
      <div className='media-icon'>
        {props.type === MediaType.Video ? (
          <VideocamOutlinedIcon />
        ) : (
          <InsertPhotoOutlinedIcon />
        )}
      </div>
      <div className='media-name'>{props.name}</div>
      <div className='media-status'>
        {MediaStatus(props.status, props.progress)}
      </div>
    </div>
  );
};

export default MediaItem;
