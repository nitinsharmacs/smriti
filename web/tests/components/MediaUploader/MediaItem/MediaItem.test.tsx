import { render } from '@testing-library/react';
import MediaItem from 'src/components/MediaUploader/MediaItem/MediaItem';
import {
  MediaType,
  MediaUploadStatus,
} from 'src/components/MediaUploader/types';
import { describe, expect, it } from 'vitest';

describe('MediaItem', () => {
  it('should render success image media item', () => {
    const { container } = render(
      <MediaItem
        name={''}
        type={MediaType.Image}
        status={MediaUploadStatus.Success}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render success video media item', () => {
    const { container } = render(
      <MediaItem
        name={''}
        type={MediaType.Video}
        status={MediaUploadStatus.Success}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render failed image media item', () => {
    const { container } = render(
      <MediaItem
        name={''}
        type={MediaType.Image}
        status={MediaUploadStatus.Failed}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render failed video media item', () => {
    const { container } = render(
      <MediaItem
        name={''}
        type={MediaType.Video}
        status={MediaUploadStatus.Failed}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render progress image media item', () => {
    const { container } = render(
      <MediaItem
        name={''}
        type={MediaType.Image}
        status={MediaUploadStatus.InProgress}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render progress video media item', () => {
    const { container } = render(
      <MediaItem
        name={''}
        type={MediaType.Video}
        status={MediaUploadStatus.InProgress}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
