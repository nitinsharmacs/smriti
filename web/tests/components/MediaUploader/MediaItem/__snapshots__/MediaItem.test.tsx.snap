// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MediaItem > should render failed image media item 1`] = `
<div>
  <div
    class="upload-media-item"
  >
    <div
      class="media-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="InsertPhotoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-4.86 8.86-3 3.87L9 13.14 6 17h12z"
        />
      </svg>
    </div>
    <div
      class="media-name"
    />
    <div
      class="media-status"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ReplayIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`MediaItem > should render failed video media item 1`] = `
<div>
  <div
    class="upload-media-item"
  >
    <div
      class="media-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="VideocamOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 8v8H5V8zm1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1"
        />
      </svg>
    </div>
    <div
      class="media-name"
    />
    <div
      class="media-status"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ReplayIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`MediaItem > should render progress image media item 1`] = `
<div>
  <div
    class="upload-media-item"
  >
    <div
      class="media-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="InsertPhotoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-4.86 8.86-3 3.87L9 13.14 6 17h12z"
        />
      </svg>
    </div>
    <div
      class="media-name"
    />
    <div
      class="media-status"
    >
      <div
        class="circular-progressbar"
        style="position: relative; width: 20px; height: 20px;"
      >
        <span
          aria-valuenow="100"
          class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary circular-progressbar-bg css-10tu4h6-MuiCircularProgress-root"
          role="progressbar"
          style="width: 20px; height: 20px; transform: rotate(-90deg);"
        >
          <svg
            class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-eorpq6-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="18"
              stroke-width="8"
              style="stroke-dasharray: 113.097; stroke-dashoffset: 0.000px;"
            />
          </svg>
        </span>
        <span
          aria-valuenow="0"
          class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary circular-progressbar-stroke css-1mgn64i-MuiCircularProgress-root"
          role="progressbar"
          style="width: 20px; height: 20px; transform: rotate(-90deg);"
        >
          <svg
            class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate MuiCircularProgress-circleDisableShrink css-eorpq6-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="18"
              stroke-width="8"
              style="stroke-dasharray: 113.097; stroke-dashoffset: 113.097px;"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MediaItem > should render progress video media item 1`] = `
<div>
  <div
    class="upload-media-item"
  >
    <div
      class="media-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="VideocamOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 8v8H5V8zm1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1"
        />
      </svg>
    </div>
    <div
      class="media-name"
    />
    <div
      class="media-status"
    >
      <div
        class="circular-progressbar"
        style="position: relative; width: 20px; height: 20px;"
      >
        <span
          aria-valuenow="100"
          class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary circular-progressbar-bg css-10tu4h6-MuiCircularProgress-root"
          role="progressbar"
          style="width: 20px; height: 20px; transform: rotate(-90deg);"
        >
          <svg
            class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-eorpq6-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="18"
              stroke-width="8"
              style="stroke-dasharray: 113.097; stroke-dashoffset: 0.000px;"
            />
          </svg>
        </span>
        <span
          aria-valuenow="0"
          class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary circular-progressbar-stroke css-1mgn64i-MuiCircularProgress-root"
          role="progressbar"
          style="width: 20px; height: 20px; transform: rotate(-90deg);"
        >
          <svg
            class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate MuiCircularProgress-circleDisableShrink css-eorpq6-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="18"
              stroke-width="8"
              style="stroke-dasharray: 113.097; stroke-dashoffset: 113.097px;"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MediaItem > should render success image media item 1`] = `
<div>
  <div
    class="upload-media-item"
  >
    <div
      class="media-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="InsertPhotoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-4.86 8.86-3 3.87L9 13.14 6 17h12z"
        />
      </svg>
    </div>
    <div
      class="media-name"
    />
    <div
      class="media-status"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`MediaItem > should render success video media item 1`] = `
<div>
  <div
    class="upload-media-item"
  >
    <div
      class="media-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="VideocamOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 8v8H5V8zm1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1"
        />
      </svg>
    </div>
    <div
      class="media-name"
    />
    <div
      class="media-status"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;
